include/deal.II/grid/tria.h

```diff
@@ -1921,6 +1920,14 @@ class Triangulation : public Subscriptor
     boost::signals2::signal<void ()> create;
     boost::signals2::signal<void ()> pre_refinement;
     boost::signals2::signal<void ()> post_refinement;
+    /**
+     * @note the signal parameter @p cell is correspond to the new active cell
+     * after coarsening is done rather than a current active cell.
+     * In another word, all children of @p cell will be deleted after coarsening
+     * and @p cell will become the new active cell then.
+     */
+    boost::signals2::signal<void (const Triangulation<dim, spacedim>::cell_iterator &cell)> pre_coarsening_on_cell;
+    boost::signals2::signal<void (const Triangulation<dim, spacedim>::cell_iterator &cell)> post_refinement_on_cell;
```

Comment:
I moved the individual discussions to `Signals` structure because at there it's more convenient to refer to parameters.